docs/tempo/website/metrics-generator/service_graphs.md

```diff
@@ -21,46 +21,49 @@ and the connections and dependencies between its components:
   Distributed systems change very frequently,
   and service graphs offer a way of seeing how these systems have evolved over time.
 
-<p align="center"><img src="../service-graphs.png" alt="Service graphs example"></p>
+<p align="center"><img src="../grafana-service-graphs-panel.png" alt="Service graphs example"></p>
 
 ## How they work
 
 The metrics-generator will process traces and generate service graphs in the form of prometheus metrics.
 
 Service graphs work by inspecting traces and looking for spans with parent-children relationship that represent a request.
-Additionally, spans need to contain the tag [`span.kind`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind),
-with value `CLIENT` for the parent span (i.e. service that started the request) and `SERVER` for the children span (i.e. service that received the request).
+The processor uses the [OpenTelemetry semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/README.md) to detect a myriad of requests.
+It currently supports the following requests:
+- a direct request between two services: the outgoing and the incoming span must have [`span.kind`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind) `client` and `server` respectively.
+- a request across a messaging system: the outgoing and the incoming span must have `span.kind` `producer` and `consumer` respectively.
+- a database request: in this case the processor looks for a single span with `span.kind` equal to `client` containing the attribute `db.name`.
 
-If these conditions are met, those spans are recorded as an edge in the graph, represented by a metric.
-The nodes in the graphs, or services, are indicated in that metric by the labels `client` and `server`.
+Every span that can be paired up to form a request is kept in an in-memory store, until its corresponding pair span is received or the maximum waiting time has passed.
+When either of these conditions are reached, the request is recorded and removed from the local store.
+
+Each emitted metrics serie will have the `client` and `server` label corresponding with the service doing the request and the service receiving the request.
 
 ```
-  tempo_service_graph_request_total{client="app", server="db"} 20
+  tempo_service_graph_request_total{client="app", server="db", connection_type="database"} 20
```

Comment:
Would it be worth adding some details about what the different `connection_type` label values could be?

Reply:
Yeah agree, I can add something under the metrics table I think.